ipatests/test_integration/test_installation_client.py

```diff
@@ -188,3 +250,219 @@ def test_client_nsupdate(self, setup_bindserver):
+
+    def test_install_invalid_domain(self):
+        """Installation with invalid domain"""
+        client = self.clients[0]
+        cmd = client.run_command(
+            "ipa-client-install "
+            "--domain=xxx.xx "
+            f"-p {client.config.admin_name} "
+            f"-w {client.config.admin_password} "
+            "-U",
+            raiseonerr=False)
+        assert cmd.returncode != 0
+        exp_msg = "Unable to find IPA Server to join"
+        assert exp_msg.lower() in cmd.stderr_text.lower()
+
+    def test_client_install_no_domain(self):
+        """
+        Installation with the server but without
+        the required param domain specified
+        """
+        client = self.clients[0]
+        cmd = client.run_command(
+            "ipa-client-install --server=xxx",
+            raiseonerr=False)
+        assert cmd.returncode != 0
+        exp_msg = "--server cannot be used without providing --domain"
+        assert exp_msg.lower() in cmd.stderr_text.lower()
+
+    def test_client_install_invalid_server(self):
+        """Installation with invalid server"""
+        client = self.clients[0]
+        cmd = client.run_command(
+            "ipa-client-install --server=xxx --domain=xxx.xx",
+            raiseonerr=False)
+        assert cmd.returncode != 0
+        exp_msg = "failed to verify that xxx is an ipa server"
+        assert exp_msg.lower() in cmd.stderr_text.lower()
+
+    def test_client_install_invalid_hostname(self):
+        cmd = tasks.install_client(
+            self.master, self.clients[0],
+            extra_args=['--hostname', 'nonexistent'],
+            raiseonerr=False
+        )
+        exp_msg = "invalid hostname"
+        assert cmd.returncode != 0
+        assert exp_msg.lower() in cmd.stderr_text.lower()
+
+    def test_client_install_password_no_principal(self):
+        cmd = self.clients[0].run_command(
+            "ipa-client-install --password=random123 -U",
+            raiseonerr=False)
+        exp_msg = "invalid credentials"
+        assert cmd.returncode != 0
+        assert exp_msg.lower() in cmd.stderr_text.lower()
+
+    def test_client_install_nonexistent_principal(self):
+        cmd = tasks.install_client(
+            self.master, self.clients[0],
+            user="nonexistentprincipal",
+            raiseonerr=False
+        )
+        exp_msg = "not found in Kerberos database"
+        assert cmd.returncode != 0
+        assert exp_msg.lower() in cmd.stderr_text.lower()
+
+    def test_client_install_non_admin_principal(self):
+        non_admin_user = "testuser"
+        tasks.create_active_user(self.master,
+                                 non_admin_user,
+                                 "RandomPassword123")
+        cmd = tasks.install_client(
+            self.master, self.clients[0],
+            user=non_admin_user,
+            raiseonerr=False
+        )
+        exp_msg = "Password incorrect while getting initial credentials"
```

Comment:
This error is unexpected. I'd expect something more like:

Joining realm failed: No permission to join this host to the IPA domain.

The password should have been properly set when the user was created so I think that that password isn't being passed into the installer.

Reply:
I am not sure about this one.

The non-admin user is created as an active one. So the password for them are set.
The client installation uses default admin password, if not provided with one explicitly.

Reply:
@rcritten do you happen to have any insight as what is happening here, please?

Reply:
You create a test user with the password of RandomPassword123. The user password is made valid by tasks.create_active_user()
You then run the client install task as non_admin_user but you don't pass in the password. tasks.install_client() defaults to the admin password if you don't provide one. It doesn't match which is why this test passes with the expected error. I suppose it might be nicer to be explicit about passing in a different password, though you don't really need a non-admin user for that I suppose.